### PR TITLE
Populate board

### PR DESCRIPTION
**Pull request by jeffp2007**
Populate board with pieces. Tests are included.

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -5,4 +5,128 @@ class Game < ActiveRecord::Base
...
   belongs_to :black_user, class_name: 'User', foreign_key: 'black_uid'
 
+  after_create :populate_board!
+  
+ 	def populate_board!
+ 		game = Game.last.id
+	
+		######create rooks######
+		0.step(7,7) do |x|
+			Piece.create!(
```
Comment (johnnyiller) [inline app/models/game.rb]: each piece should be responsible for initializing itself.  The populate board should be concerned with creating new pieces associated with the proper user and the proper game, but the initial position should be encapsulated in the pieces class.  Additionally you should not be creating a "Piece" but create a "Rook"...  Try re-doing this, this time using polymorphism to initialize the pieces.

File: test/models/game_test.rb [hunk 2]
```diff
@@ -4,4 +4,308 @@ class GameTest < ActiveSupport::TestCase
...
+
+
+  #####################################################################################################
+	################################# POPULATE BLACK PIECES #############################################
+	#####################################################################################################
+
+
+  ##### TEST POPULATE 2 BLACK ROOKS ######
+
+	test "populate black rook - 1 of 2" do
```
Comment (johnnyiller) [inline test/models/game_test.rb:16]: excellent work on the tests this is great keep it up.

File: test/models/game_test.rb [hunk 3]
```diff
@@ -4,4 +4,308 @@ class GameTest < ActiveSupport::TestCase
...
+		game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
+		expected = "Bishop0"
+		bishop = game.pieces.where(:pos_x => 2).where(:pos_y => 0).first
+		actual = bishop.type.to_s + bishop.color.to_s
+		assert_equal expected, actual
+	end
+
+	test "populate black bishop - 2 of 2" do
+		game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
+		expected = "Bishop0"
```
Comment (johnnyiller) [inline test/models/game_test.rb]: The only thing i'd say is that this is a little strange.  There is no reason you can't assert 2 things in a test.  So for example you can have

assert_equal "Bishop", bishop.type.to_s
assert_equal 0, bishop.color

it makes the test read more clearly for future users.  That said, you don't need to change it but just remember for next time.

File: app/models/game.rb (hunk 1 above)
Comment (jeffp2007) [inline app/models/game.rb]: Ah, now I am lost.

Comment (cryogenx) [inline app/models/game.rb]: something like: [This Gist](https://gist.github.com/cryogenx/d23c12245b4c33cb8faa#file-pawns-rb)
 or

``` ruby
    # Populate Pawns
    (0..7).each do |x|
      Pawn.create(pos_x: x, pos_y: 1, game_id: self.id, color: 1, status: 'active') # White
      Pawn.create(pos_x: x, pos_: 6, game_id: self.id, color: 0,status: 'active') # Black
    end
```

Comment (cryogenx) [inline app/models/game.rb]: for the rooks:

``` ruby
    Rook.create(pos_x: 0, pos_y: 0, game_id: self.id, color: 1, status: 'active') # White
    Rook.create(pos_x: 7, pos_y: 0, game_id: self.id, color: 1, status: 'active') # White
    Rook.create(pos_x: 0, pos_y: 7, game_id: self.id, color: 0, status: 'active') # Black
    Rook.create(pos_x: 7, pos_y: 7, game_id: self.id, color: 0, status: 'active') # Black
```

**Comment by jeffp2007** [comment]
This is as far as I could take it. I couldn't figure out how to create pieces in each of the individual models after hours of research, trial and error.

**Comment by cryogenx** [comment]
I'm confused I though you should only merge if the feature was ready for production?

**Comment by johnnyiller** [comment]
It's true that you should only merge if something is ready for production, which this code is in many respects.  It might not be the way I would implement it, but the code has tests that verify that it is working.  Technically speaking this is how "Agile" software development works.  You can push anything to production that isn't broken and simply fix the implementation over time.  This is why the tests are important, without them it would be difficult to look back at this code and make improvements.

**Comment by cryogenx** [comment]
Got it..

**Comment by jeffp2007** [comment]
Thanks for the insight @johnnyiller

**Comment by johnnyiller** [comment]
no problem

**Pull request merged**